app/models/manageiq/providers/hawkular/middleware_manager.rb

```diff
@@ -498,18 +533,24 @@ def run_specific_operation(operation_name, ems_ref, parameters = {})
       run_operation(parameters, operation_name)
     end
 
-    def run_operation(parameters, operation_name = nil)
+    def run_operation(parameters, operation_name = nil, extra_data = {})
       with_provider_connection do |connection|
         callback = proc do |on|
-          on.success do |data|
-            _log.debug "Success on websocket-operation #{data}"
-            emit_middleware_notification(:mw_op_success, parameters[:operationName], nil, parameters[:resourcePath],
-                                         MiddlewareServer)
+          notification_args = EmitNotificationArgs.new(
```

Comment:
This is good, but all the responsibilities on mapping the data correctly are still on the caller. `extra_data` and `parameters` could be sent to the object directly, and it would get the correct key from there. I've been thinking something like this would be a good interface, which would move the two conditional and data introspection to the data object:

```ruby
EmitNotificationArgs.new(:mw_op_success, MiddlewareServer, parameters, extra_data)
```

Also, for the name, what do you think of just `NotificationArgs`?

Reply:
Ok, I'll use `NotificationArgs`.
I'll skip the suggestion of moving the mapping responsibility, because this is the only usage where it makes sense and breaks all other usages.